### PR TITLE
Allow dynamic datasets and namespaces

### PR DESCRIPTION
**Pull request by dmathieu**
## Motivation/summary

This change grants the permissions for agents running the apmpackage integration to send data to other data streams, and other namespaces, making the use of the `reroute` processor possible.

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [x] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
- [x] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes

I ran this integration locally with tilt.

## Related issues

Closes https://github.com/elastic/apm-server/issues/10991

### CODE REVIEW

File: apmpackage/apm/changelog.yml [hunk 1]
```diff
@@ -3,6 +3,9 @@
     - description: Placeholder
```
Comment (dmathieu) [inline apmpackage/apm/changelog.yml:3]: Should the placeholder be removed manually, or do we remove it when the release happens?

Comment (simitt) [reply apmpackage/apm/changelog.yml:3]: Afair, it's removed when we branch out.

**Review by simitt — COMMENTED** [review]
From the issue description:
> to add elasticsearch.dynamic_dataset: true only to the [app_metrics](https://github.com/elastic/apm-server/tree/main/apmpackage/apm/data_stream/app_metrics) and [app_logs](https://github.com/elastic/apm-server/tree/main/apmpackage/apm/data_stream/app_logs) data streams. That's because changing the data set for other data stream, such as [internal_metrics](https://github.com/elastic/apm-server/tree/main/apmpackage/apm/data_stream/internal_metrics) could confuse the APM UI.

Was this overlooked or were there any follow up conversations/considerations that led to adding `dynamic_dataset: true` everywhere?

**Comment by dmathieu** [comment]
> Was this overlooked or were there any follow up conversations/considerations that led to adding dynamic_dataset: true everywhere?

Ah yes, my bad. I overlooked that. I've kept it only for app_metrics and app_logs.
I do wonder if we should would be interesting to have it for traces as well?

**Comment by simitt** [comment]
I was wondering the same tbh. @felixbarny was there a special reason for keeping `traces-*` out of this? Any concerns for the APM UI?

**Comment by felixbarny** [comment]
I don't have a strong opinion on whether to allow dynamic_dataset for traces. I suppose there's no harm in allowing that.

However, we should add `dynamic_namespace` to all APM data streams. This enables users to send all APM data to a dedicated namespace so that they can use custom retention and security policies.

**Review by axw — APPROVED** [review]
LGTM. If we go ahead with https://github.com/elastic/elasticsearch/pull/97546, then I expect we'll end up giving APM append-only privileges for `traces-*`, `metrics-*`, and `logs-*`. This seems pretty well aligned.

**Comment by axw** [comment]
Verified with 8.10.0-BC1.

I created `@custom` ingest pipelines with the `reroute` processor, setting dataset to a constant value ("apm.app.all") for app logs/metrics; and setting namespace to a template (some "{{agent.name}}", some "{{service.name}}")

Sending test data with apm-tool, I verified this with `logs-apm.app.*`, `metrics-apm.app.*`, `traces-apm-*`, `logs-apm.error-*`, and `metrics-apm.service_destination.1m-*`.

**Pull request merged**